Documentation/how-to-build-and-run-ilcompiler-in-console-shell-prompt.md

```diff
@@ -53,3 +54,18 @@ From the shell/command prompt, issue the following commands to generate the nati
     dotnet restore
     dotnet compile --native --cpp --ilcpath bin\Product\Windows_NT.x64.Debug\.nuget\publish1
 ```
+
+## Workarounds for linker errors on Windows ##
+
+If you are seeing errors such as: 
+
+```
+    LINK : fatal error LNK1104: cannot open file 'kernel32.lib'
+    Linking of intermediate files failed.
+```
+
+There are a few workarounds you might try:
+ - Make sure you run these commands from a VS developer X64 command prompt instead of a vanilla command prompt
+ - Search for the missing lib files in your SDK, for example under C:\Program Files (x86)\Windows Kits\10\lib. Make sure the path to these libraries is included in the LIB environment variable. It appears VS 2015 RTM developer command prompt does not correctly set the LIB paths for the 10586 Windows SDK. VS 2015 Update 1 resolves that issue, so installing it is another alternative.
```

Comment:
I attempted to recreate my repro but I was unsuccesful. Installing VS2015 RTM + 10.10586 SDK gave correct LIB paths for both VS2015 x64 Native Tools and Developer Command Prompt for 2015. Looks like Jan already merged the change. I have no complaints if you want to remove the workaround instructions, or if we leave them there for a while just in case.

I’m guessing at some point the dotnet native compile won’t have a dependency on having a Windows SDK installed, in which case that seems like a good time to also remove the workaround portion of the instructions.

-Noah

From: Gaurav Khanna [mailto:notifications@github.com]
Sent: Tuesday, January 12, 2016 7:38 AM
To: dotnet/corert <corert@noreply.github.com>
Cc: Noah Falk <noahfalk@microsoft.com>
Subject: Re: [corert] Update build instructions to address linker errors (#625)


In Documentation/how-to-build-and-run-ilcompiler-in-console-shell-prompt.md<https://github.com/dotnet/corert/pull/625#discussion_r49469384>:

> @@ -53,3 +54,18 @@ From the shell/command prompt, issue the following commands to generate the nati

>      dotnet restore

>      dotnet compile --native --cpp --ilcpath bin\Product\Windows_NT.x64.Debug\.nuget\publish1

>  ```

> +

> +## Workarounds for linker errors on Windows ##

> +

> +If you are seeing errors such as:

> +

> +```

> +    LINK : fatal error LNK1104: cannot open file 'kernel32.lib'

> +    Linking of intermediate files failed.

> +```

> +

> +There are a few workarounds you might try:

> + - Make sure you run these commands from a VS developer X64 command prompt instead of a vanilla command prompt

> + - Search for the missing lib files in your SDK, for example under C:\Program Files (x86)\Windows Kits\10\lib. Make sure the path to these libraries is included in the LIB environment variable. It appears VS 2015 RTM developer command prompt does not correctly set the LIB paths for the 10586 Windows SDK. VS 2015 Update 1 resolves that issue, so installing it is another alternative.

Once we are in "Native Tools for x64" command prompt, these errors shouldn't exist. Can you please confirm if you still see them?

—
Reply to this email directly or view it on GitHub<https://github.com/dotnet/corert/pull/625/files#r49469384>.